listenbrainz/webserver/static/js/src/APIService.ts

```diff
@@ -969,4 +969,23 @@ export default class APIService {
     await this.checkStatus(response);
     return response.json();
   };
+
+  deleteFeedEvent = async (
+    eventType: string,
+    username: string,
+    userToken: string,
+    id?: number
```

Comment:
I don't know how it is implemented on the server but if an ID is required then this method should probably throw an error if there is no `id`
It would be great to add a test case for that as well.

Reply:
hi @MonkeyDo, really sorry for late reply because of exams,
so, id is indeed implemented in the backend
https://github.com/metabrainz/listenbrainz-server/pull/1803/files#diff-68b3baf8382d61b3b29a2894bb9edca6db7a948d6f0794bf28275c8581f859b4R462

When I started working on LB-912, we realised that ID needs to be passed in multiple APIs, hence I implemented id to be an optional parameter in #1610 .
https://github.com/metabrainz/listenbrainz-server/pull/1610/files#diff-c99f9398509233652fe3be847b91ec0e6b89d968b4bdc68cf81063767aed7683R80

so since `id` is indeed being passed by the backend for all those events that can be deleted from the feed page, hence, it shouldn't be a problem.

Reply:
No problem, take your time! Your exams are a lot more important, no question about that :)

Regarding the optional ID, as far as I can tell the argument for this API method should be `id: number`, i.e. not optional, as it doesn't make sense to call `/feed/events/delete` without an id.

I'd still like you to cover the case of a missing id parameter and throw an error with a helpful message.
Otherwise, if something changes in the future and we call this method without the id, something is likely to break in less clear ways.

Reply:
Hi, this totally makes sense, the embarrassing thing is that I mistakenly thought that this is a code review for `types.d.ts` and initially thought you wanted me to remove `?` from `id?: number` from the same file (since I had no recollection of putting `id?: number` in `deleteFeedEvent`). But now I do remember that I had put it because I was facing some typescript error and then forgot to switch it back and it completely vanished from my mind. ;_; , which is why I deleted my earlier comment :P

also, the edge case of `id` not being there for future changes is a very good idea, hadn't crossed my mind before. I would implement that for sure. Thank you so much @MonkeyDo!!

Reply:
Let me know if you need a hand with the typescript issues :)